query/sources/impl/src/test/groovy/org/codice/ddf/admin/sources/SourceTestCommons.groovy

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.sources
+
+import ddf.catalog.data.ContentType
+import ddf.catalog.operation.QueryRequest
+import ddf.catalog.operation.SourceResponse
+import ddf.catalog.service.ConfiguredService
+import ddf.catalog.source.Source
+import ddf.catalog.source.SourceMonitor
+import ddf.catalog.source.UnsupportedQueryException
+import org.codice.ddf.admin.common.fields.common.*
+import org.codice.ddf.admin.common.services.ServiceCommons
+import org.codice.ddf.admin.sources.fields.type.SourceConfigUnionField
+
+class SourceTestCommons {
+
+    static final ENDPOINT_URL = SourceConfigUnionField.ENDPOINT_URL_FIELD_NAME
+
+    static final CREDENTIALS = CredentialsField.DEFAULT_FIELD_NAME
+
+    static final USERNAME = CredentialsField.USERNAME_FIELD_NAME
+
+    static final PASSWORD = CredentialsField.PASSWORD_FIELD_NAME
+
+    static final HOST = HostField.DEFAULT_FIELD_NAME
+
+    static final ADDRESS = AddressField.DEFAULT_FIELD_NAME
+
+    static final PORT = PortField.DEFAULT_FIELD_NAME
+
+    static final HOSTNAME = HostnameField.DEFAULT_FIELD_NAME
+
+    static final URL_NAME = UrlField.DEFAULT_FIELD_NAME
+
+    static final PID = PidField.DEFAULT_FIELD_NAME
+
+    static final FACTORY_PID_KEY = ServiceCommons.FACTORY_PID_KEY
+
+    static final SERVICE_PID_KEY = ServiceCommons.SERVICE_PID_KEY
+
+    static final SOURCE_NAME = SourceConfigUnionField.SOURCE_NAME_FIELD_NAME
+
+    static final SOURCE_CONFIG = SourceConfigUnionField.FIELD_NAME
+
+    static final ID = 'id'
+
+    static final FLAG_PASSWORD = ServiceCommons.FLAG_PASSWORD
+
+    static F_PID = "testFactoryPid"
+
+    static S_PID = "testServicePid"
+
+    static S_PID_1 = "testServicePid1"
+
+    static S_PID_2 = "testServicePid2"
+
+    static SOURCE_ID_1 = "testId1"
+
+    static SOURCE_ID_2 = "testId2"
+
+    static TEST_USERNAME = "admin"
+
+    static TEST_PASSWORD = "admin"
+
+    static TEST_SOURCENAME = "testSourceName"
+
+    static discoverByAddressActionArgs = [
```

Comment:
We should be creating the fields and calling getValue on them instead of creating our own maps. This will ensure our objects are transforming properly.